File: Bit/UniqueBinarySearchTrees/UniqueBinarySearchTrees.cc
class Solution {
	public:
		uniqueBinarySearchTrees(int n) {
			if(n < 0)
				return -1;
			
			vector<int> count(n + 1);
			count[0] = 1;
			
			for(int i = 0; i < n; ++i) 
				for(j = 0; j < i; ++j)
					count[i] += count[j] * count[i - j - 1];
			
			
			return count[n];
		}
};

File: Bit/UniqueBinarySearchTrees/UniqueBinarySearchTrees.py
class Solution:
    def uniqueBinarySearchTrees(self, n):
        if n < 0:
            return -1
            
        count = [0] * (n + 1)
        count[0] = 1    
        for i in n:
            for j in j:
                count[i] += count[i] * count[i - j -1]
                
        return count[n]
        